core/application/reconcile.go

```diff
@@ -162,8 +160,11 @@ func (a *app) reconcile(ctx context.Context, spec *models.MicroVM, logger *logru
 		return nil
 	}
 
-	spec.Status.Retry = 0
-	spec.Status.NotBefore = 0
+	return a.saveState(ctx, spec, plan, models.CreatedState)
+}
+
+func (a *app) saveState(ctx context.Context, spec *models.MicroVM, plan planner.Plan, state models.MicroVMState) error {
+	plan.Finalise(state)
```

Comment:
Oh, is it even used?

```
func (a *app) saveState(ctx context.Context, spec *models.MicroVM, plan planner.Plan, state models.MicroVMState) error {
	plan.Finalise(state)

	if _, err := a.ports.Repo.Save(ctx, spec); err != nil {
		return fmt.Errorf("saving spec after plan execution: %w", err)
	}
	return nil
}
```

`Finalise` sets `p.vm.Status.Retry` not `*spec`, but we save `*spec`

Reply:
explained on call, but recapping here for posterity:

The status which is altered in Plan _is_ the one which is saved in Repo. We have a not-deal situation in the whole plan/steps/service wiring, where everyone gets a pointer to the spec and literally everyone is changing the status on that pointer. So it does get saved, but it is a pain to follow through mentally.
It would be good to have a refactor at some point to untangle it and make it a little more linear/easier to parse, but for now you are gonna have to take my word that all the changes which are made elsewhere to that spec are saved on it 😄 .